### PR TITLE
BracesFixer - Removes line break

### PR DESCRIPTION
**Pull request by SpacePossum**
Bug on master only
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1755

### CODE REVIEW

**Comment by SpacePossum** [comment]
ping @GrahamCampbell and @keradus

**Comment by GrahamCampbell** [comment]
Looks good. :+1:

File: src/Tokenizer/TokensAnalyzer.php [hunk 1]
```diff
@@ -564,19 +564,13 @@ public function isWhilePartOfDoWhile($index)
...
         }
 
+        $startParenthesisIndex = $tokens->getNextMeaningfulToken($index);
         $endParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startParenthesisIndex);
-
-        $nextIndex = $tokens->getNextMeaningfulToken($endParenthesisIndex);
-        $nextToken = $tokens[$nextIndex];
-
-        if ($nextToken->equals(';')) {
+        if ($tokens[$tokens->getNextMeaningfulToken($endParenthesisIndex)]->equals(';')) {
```
Comment (SpacePossum) [inline src/Tokenizer/TokensAnalyzer.php]: will fail for 
```php
<?php
$i = 1;
do {
    ++$i;
    
} while($i <5)?>
```

.... sigh oh man...

File: tests/Tokenizer/TokensAnalyzerTest.php [hunk 2]
```diff
@@ -670,4 +670,49 @@ class TestClass {
 
         return $cases;
     }
+
+    /**
+     * @dataProvider provideIsWhilePartOfDoWhile
+     */
+    public function testIsWhilePartOfDoWhile($source, $index, $expected)
```
Comment (SpacePossum) [inline tests/Tokenizer/TokensAnalyzerTest.php]: TODO missing test for exception when index passed for token not of kind `T_WHILE`

**Comment by keradus** [comment]
Please rebase

**Comment by SpacePossum** [comment]
rebase done

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**